routes/project/location/private_subnet.rb

```diff
@@ -84,7 +84,18 @@ class Clover
 
         r.delete do
           authorize("PrivateSubnet:delete", ps.id)
-          unless ps.vms.all? { it.destroy_set? || it.strand.nil? || it.strand.label == "destroy" }
+
+          vms_dataset = ps.vms_dataset
+            .association_join(:strand)
+            .exclude(label: "destroy")
+            .exclude(Sequel[:vm][:id] => Semaphore
+              .where(
+                strand_id: DB[:nic].where(private_subnet_id: ps.id).select(:vm_id),
```

Comment:
Why don't we use `ps.nics_dataset` directly?

Reply:
That makes sense. As there is a PR depending on this one (already approved), I think I'll wait until around merging time to change over.